app/agents/voice/automatic/tools/breeze/utils.py

```
@@ -526,3 +526,263 @@ def validate_and_process_surcharge_rules(
+            )
+            logger.info(
+                f"Optimization complete: {len(backup['rules'])} → 0 rules (all rules eliminated)"
+            )
+            return True, [], backup
+
+        logger.info(
+            f"Optimization complete: {len(backup['rules'])} → {len(final_rules)} rules"
+        )
+        return True, final_rules, backup
+
+    except Exception as e:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)